components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/validation/ValidatorExt.java

```diff
@@ -0,0 +1,33 @@
+/*
+*  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.dataservices.core.validation;
+
+import java.util.Map;
+
+/**
+ * This interface represents a custom validator with property support for data services fields.
+ */
+public interface ValidatorExt extends Validator {
+
+    /**
+     * method used to set properties for the custom validator class
```

Comment:
Start a line with a capital letter, End with a full stop. The main comment should follow these rules.